app/persistence/valkyrie/persistence/solr/queries/find_references_query.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+module Valkyrie::Persistence::Solr::Queries
+  class FindReferencesQuery
+    attr_reader :model, :property, :connection, :resource_factory
+    def initialize(model:, property:, connection:, resource_factory:)
+      @model = model
+      @property = property
+      @connection = connection
+      @resource_factory = resource_factory
+    end
+
+    def run
+      enum_for(:each)
+    end
+
+    def each
+      unordered_result.each do |member|
+        yield member
+      end
+    end
+
+    def unordered_result
+      docs.map do |doc|
+        resource_factory.to_model(doc)
+      end
+    end
+
+    def docs
+      connection.get("select", params: { q: query, rows: 1_000_000_000 })["response"]["docs"]
```

Comment:
Any particular reason this query uses `1_000_000_000` vs. `100_000` in `app/persistence/valkyrie/persistence/fedora/queries/find_references_query.rb`?  I doubt it would make a difference (I suspect 100K records would overwhelm just as well as 1M would)...